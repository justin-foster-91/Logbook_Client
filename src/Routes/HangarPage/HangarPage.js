import React, { useEffect, useState } from 'react';
import './HangarPage.css';
import HangarDisplay from '../../Components/HangarDisplay/HangarDisplay';
import ShipApiService from '../../Services/ship-api-service';

const HangarPage = (props) => {
  const [ships, setShips] = useState([])

  const clickNewShip = () => {
    ShipApiService.createShip()
      .then(newShip => {
        console.log("clickNewShip -> newShip", newShip)
        console.log("clickNewShip -> ships", ships)
        ships.push(newShip)
        console.log("clickNewShip -> newShip", newShip)
        console.log("clickNewShip -> ships", ships)
        setShips(ships)
      })
      .catch(e => console.log(e))
  }
  return (
    <div>
      <h1>
        {"Hello Hangar Page"}
      </h1>
      <HangarDisplay setShips={setShips} ships={ships}/>
      <button onClick={() => clickNewShip()}>Create New Ship</button>
    </div>
  );
  
}

export default HangarPage;
