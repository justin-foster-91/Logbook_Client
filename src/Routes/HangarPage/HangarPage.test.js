import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import HangarPage from './HangarPage';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BrowserRouter><HangarPage /></BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
})