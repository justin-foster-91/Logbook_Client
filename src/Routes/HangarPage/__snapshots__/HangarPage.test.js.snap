// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <h1>
    My Personal Hangar
  </h1>
  <button
    onClick={[Function]}
  >
    Create New Ship
  </button>
</div>
`;
